gateway/proxy_muxer.go

```diff
@@ -406,14 +408,14 @@ func (m *proxyMux) serve() {
 			}
 			var h http.Handler
 			h = &handleWrapper{p.router}
-			if p.protocol == "h2c" {
-				// wrapping handler in h2c. This ensures all features including tracing work
-				// in h2c services.
-				h = &h2cWrapper{
-					w: h.(*handleWrapper),
-					h: h2c.NewHandler(h, &http2.Server{}),
-				}
+			// by default enabling h2c by wrapping handler in h2c. This ensures all features including tracing work
```

Comment:
This is a very big assumptions. By default we provide http proxy, so h2c,tcp etc are special case. You explicitly set protocol to h2c inside loadApps why not check it here?. I see this will be a major source of headache.

Reply:
@gernest hi, can you elaborate regarding why it could be a major source of headache? currently by enabling h2c by default we're looking to simplify the process the most that we can

Reply:
@sredxny 
By design `http2` is supposed to be served over TLS,  `h2c` is an edge case/optional that is not supported by major browsers.

The only use case for `h2c` is when using `gRPC` without tls, which is also an edge case. Why do we make edge cases default behaviour? Like I said before http/http2 is the only thing we should take for granted, we should let our customers to choose anything else

Please see this stackoverflow answer on why http2 should always be on TLS https://stackoverflow.com/a/46789195

>currently by enabling h2c by default we're looking to simplify the process the most that we can

What process are you simplifying? A process of choosing protocol to use? Why this  `protocol: "h2c"` is not simple?

Reply:
protocol field require you to have separate gateway port, while in general there is no reasons for this.

Reply:
So the issue here is to remove the need for separate port?

Reply:
In general yes. The whole idea of h2c that it works though standard http protocol, and if special header found, it enable special mode which kinda emulate http2 by using http as transport.